File: app/src/main/java/com/cyclecare/app/domain/repository/PeriodRepository.kt
package com.cyclecare.app.domain.repository

import com.cyclecare.app.domain.model.CycleInsights
import com.cyclecare.app.domain.model.CyclePrediction
import com.cyclecare.app.domain.model.Period
import kotlinx.coroutines.flow.Flow
import java.time.LocalDate

interface PeriodRepository {
    fun getAllPeriods(): Flow<List<Period>>
    suspend fun getPeriodById(id: Long): Period?
    suspend fun insertPeriod(period: Period): Long
    suspend fun updatePeriod(period: Period)
    suspend fun deletePeriod(period: Period)
    suspend fun predictNextCycle(): CyclePrediction?
    suspend fun getCycleInsights(): CycleInsights?
    suspend fun getPeriodsInRange(startDate: LocalDate, endDate: LocalDate): List<Period>
}

File: cyclecare_flutter/lib/domain/repositories/period_repository.dart
import '../entities/period.dart';

abstract class PeriodRepository {
  Future<List<Period>> getAllPeriods();
  Future<Period?> getPeriodById(int id);
  Stream<List<Period>> watchAllPeriods();
  Future<int> insertPeriod(Period period);
  Future<bool> updatePeriod(Period period);
  Future<int> deletePeriod(int id);
  Future<List<Period>> getPeriodsInRange(DateTime start, DateTime end);
  Future<Period?> getLastPeriod();
}
